pyproject.toml

```diff
@@ -33,10 +35,10 @@ requires-python = '>=3.7'
 
 [project.optional-dependencies]
 dev = [
-    'pre-commit~=2.2',
-    'pylint~=2.5.0',
+    'pre-commit~=3.0',
+    'pylint~=3.2.0',
     'pgtest>=1.3.1',
-    'pytest',
+    'pytest~=8.3.0',
```

Comment:
Pytest 8.3 is the last version to support Python 8. Also I think its better to pin this to prevent CI breakage.